File: es1/es1b.py
#!/usr/bin/env python

with open("input.txt", "r") as f:
    rows = f.read().split("\n")

def generator():
    while True:
        for drift in rows:
            try:
                yield int(drift)
            except ValueError:
                pass

freq = 0
m = {0: True}
for drift in generator():
    freq += int(drift)
    if freq in m:
        break
    m[freq] = True

print("Final frequency", freq)

File: es1/es1b.go
package main

import (
	"fmt"
	"os"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()

	freq := 0
	var drift int
	m := make(map[int]bool)
	for {
		_, err := fmt.Fscanf(input, "%d\n", &drift)
		if err != nil {
			input.Seek(0, 0)
			continue
		}
		freq += drift
		_, found := m[freq]
		if found {
			break
		}
		m[freq] = true
	}
	fmt.Println("Final frequency", freq)
}
